dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/call/BiStreamServerCallListener.java

```diff
@@ -27,24 +27,32 @@ public class BiStreamServerCallListener extends AbstractServerCallListener {
 
     private StreamObserver<Object> requestObserver;
 
-    public BiStreamServerCallListener(RpcInvocation invocation, Invoker<?> invoker,
-        ServerCallToObserverAdapter<Object> responseObserver) {
+    public BiStreamServerCallListener(RpcInvocation invocation,
+                                      Invoker<?> invoker,
+                                      ServerCallToObserverAdapter<Object> responseObserver) {
         super(invocation, invoker, responseObserver);
         invocation.setArguments(new Object[]{responseObserver});
         invoke();
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public void onReturn(Object value) {
-        this.requestObserver = (StreamObserver<Object>) value;
+        if (value instanceof Throwable) {
+            responseObserver.onError((Throwable) value);
+        } else if (value instanceof StreamObserver) {
+            this.requestObserver = (StreamObserver<Object>) value;
+        }
     }
 
     @Override
     public void onMessage(Object message) {
         if (message instanceof Object[]) {
             message = ((Object[]) message)[0];
         }
-        requestObserver.onNext(message);
+        if (requestObserver != null) {
```

Comment:
Making a null judgment here doesn't seem to solve the underlying problem

Reply:
Yes, just bounds checked.